cognite_toolkit/_cdf_tk/resource_classes/transformations.py

```diff
@@ -0,0 +1,151 @@
+from typing import Literal
+
+from pydantic import Field, ValidationError, model_validator
+
+from .base import ToolkitResource
+
+
+class DataModelInfo(ToolkitResource):
+    space: str = Field(description="Space of the Data Model.")
+    external_id: str = Field(description="External ID of the Data Model.")
+    version: str = Field(description="Version of the Data Model.")
+    destination_type: str = Field(description="External ID of the type(view) in the data model.")
+    destination_relationship_from_type: str | None = Field(
+        default=None, description="Property Identifier of the connection definition in destinationType."
+    )
+
+
+class ViewInfo(ToolkitResource):
+    space: str = Field(description="Space of the view.")
+    external_id: str = Field(description="External ID of the view.")
+    version: str = Field(description="Version of the view.")
+
+
+class EdgeType(ToolkitResource):
+    space: str = Field(description="Space of the type.")
+    external_id: str = Field(description="External ID of the type.")
+
+
+class DestinationSourceType(ToolkitResource):
+    type: Literal[
+        "assets",
+        "events",
+        "asset_hierarchy",
+        "datapoints",
+        "string_datapoints",
+        "timeseries",
+        "sequences",
+        "files",
+        "labels",
+        "relationships",
+        "data_sets",
+        "instances",
+        "nodes",
+        "edges",
+        "raw",
+        "sequence_rows",
+    ] = Field(description="The type of the destination resource.")
+
+
+class DataModelSource(DestinationSourceType):
+    data_model: DataModelInfo = Field(description="Target data model info.")
+    instance_space: str | None = Field(default=None, description="The space where the instances will be created.")
```

Comment:
Documentation is unclear here, but this is required
```suggestion
    instance_space: str = Field(default=None, description="The space where the instances will be created.")
```